packages/server-core/src/hooks/set-in-context.ts

```diff
@@ -31,9 +31,16 @@ import { HookContext } from '../../declarations'
  * following setField hook.
  * https://hooks-common.feathersjs.com/hooks.html#setfield
  */
-export default (propertyName: string, propertyValue: string, inData?: false) => {
+
+export enum ContextScope {
```

Comment:
this abstraction is a little worrying, what exactly is this doing?

Reply:
So, `setInContext` hook is required to to manipulate data in context.params.query, context.data or root context object. This enum allow to easily set that data from a hook object without writing custom code.

i.e. `setInContext('type', 'public'))`  will set context.params.query.type = 'public'

`setInContext('type', 'public', ContextScope.Data))`  will set context.data.type = 'public'